jobs.groovy

```diff
@@ -473,7 +473,13 @@ def list_jobs(dashProject, dashBranchName) {
                                  module: module, values: job_values,
                                  full_name: full_name])
                 }
-            } else {
+            } else if (job_values.module) {
+                full_name = full_job_name(
+                    dashProject, dashBranchName, job_name)
+                results.add([type: job_type, name: job_name,
+                             module: job_values.module, values: job_values,
+                             full_name: full_name])
```

Comment:
Allow a new type of definition that uses "module" rather than a list of
"with_modules". The difference is that this doesn't do the name mangling
that with_modules does, but still puts $MODULE in the environment.

This allows us to define the stanzas for test_docker, while keeping the same
names as they have now, and using the same body for the jobs, as they
rely on running trial ${MODULE}.